src/glarea.rs

```diff
@@ -0,0 +1,64 @@
+//! # GLArea Sample
+//!
+//! This sample demonstrates how to use GLAreas and OpenGL
+
+extern crate gtk;
+
+#[cfg(feature = "opengl")]
+mod example {
+    extern crate gl;
+    extern crate glutin;
+
+    use gtk;
+    use gtk::traits::*;
+    use gtk::signal::Inhibit;
+    use gtk::{GLArea, Window};
+
+    pub fn main() {
+        if gtk::init().is_err() {
+            println!("Failed to initialize GTK.");
+            return;
+        }
+
+        let window = Window::new(gtk::WindowType::Toplevel).unwrap();
+        let glarea = GLArea::new().unwrap();
+
+        // Hack to set up gl
+        let dummy_win = glutin::WindowBuilder::new().with_visibility(false).build().unwrap();
+        gl::load_with(|s| dummy_win.get_proc_address(s));
```

Comment:
Could you please explain briefly in a comment why this hack works, when it might not, whether it's portable and any drawbacks you're aware of?